crates/polars-ops/src/chunked_array/strings/namespace.rs

```diff
@@ -357,6 +358,38 @@ pub trait Utf8NameSpaceImpl: AsUtf8 {
         Ok(out)
     }
 
+    /// Count all successive non-overlapping regex matches.
+    fn count_match_many(&self, pat: &Utf8Chunked) -> PolarsResult<UInt32Chunked> {
+        let ca = self.as_utf8();
+        polars_ensure!(
+            ca.len() == pat.len(),
+            ComputeError: "pattern's length: {} does not match that of the argument series: {}",
+            pat.len(), ca.len(),
+        );
+
+        // Very simple cache: don't recompile the same regex for multiple values in a row.
+        // TODO: proper LRU cache with reasonable memory limit.
+        let mut reg = Regex::new("").unwrap();
+        let mut lastpat = "";
+
+        let out: UInt32Chunked = ca
```

Comment:
Can you use `try_binary_elementwise` here. The iterators have a double indirection and therefore are slower. `try_binary_elementwise` should be less boilerplante and more performant.

See https://github.com/pola-rs/polars/blob/3ea010334256fc9ccb5c13d01cb94f0e88b57285/crates/polars-core/src/chunked_array/ops/arity.rs#L47

Reply:
Thanks, I've updated the PR. I had to remove the pattern `caching` since it wouldn't satisfy the trait bounds (`FnMut`). I might be wrong however.

Reply:
@wdoppenberg What error did you get? Does it work if you do `let mut op = |...| {...}` and then `try_binary_elementwise(ca, pat, &mut op)`?

Reply:
It's the following error:
```
error[E0521]: borrowed data escapes outside of closure
   --> XXX/polars/crates/polars-ops/src/chunked_array/strings/namespace.rs:379:25
    |
372 |         let mut last_pat = "";
    |             ------------ `last_pat` declared here, outside of the closure body
373 |
374 |         let op = move |opt_s: Option<&str>, opt_pat: Option<&str>| -> PolarsResult<Option<u32>> {
    |                                             ------- `opt_pat` is a reference that is only valid in the closure body
...
379 |                         last_pat = pat.clone();
    |                         ^^^^^^^^^^^^^^^^^^^^^^ `opt_pat` escapes the closure body here

For more information about this error, try `rustc --explain E0521`.
```

I've tried the approaches you mentioned. To be fair, I might just not see how to solve it at this time.

Reply:
I seem to have found the solution. Pushing now.

Reply:
I see the lifetime problem. Let me take a look.